### ISSUE TITLE
WLM Nickname Colors

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#78]
```
What steps will reproduce the problem?
1. WLM Plus Nicknames
2. The colors of the nick names are Code, and if you dont got WLM , all you can 
see its some 
codes
3. Example ·$1,3 ·#  INF·$1,8HEC·$1,4TOR  ·$1,0 

What is the expected output? What do you see instead?
To be able to see the Colors

What version of the product are you using? On what operating system?

On Adium, OS X leopard 10.5.4
Please provide any additional information below.


```

Original issue reported on code.google.com by `calemind` on 7 Jul 2008 at 3:22

Attachments:
- [colorful_msn.jpg](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-78/comment-0/colorful_msn.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Are these supported by the official client?
```

Original comment by `felipe.contreras` on 7 Jul 2008 at 2:28
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
No men, remember that most of the features in WLM for mac, does'nt even exist!
```

Original comment by `calemind` on 7 Jul 2008 at 4:55

Attachments:
- [Image 2.png](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-78/comment-2/Image 2.png)

**Comment by GoogleCodeExporter** [comment]
```
But in WLM for Windows yes, its built in.
```

Original comment by `calemind` on 7 Jul 2008 at 4:56

**Comment by GoogleCodeExporter** [comment]
```
That's what I meant. If it's supported by the official client then it's ok.
```

Original comment by `felipe.contreras` on 8 Jul 2008 at 12:28

**Comment by GoogleCodeExporter** [comment]
```
There is a patch for Adium that add the support for this... felipec, maybe you 
can
take a look at it and try to convert it to msn-pecan. This would be great for 
pidgin
users.
http://trac.adiumx.com/ticket/6578
```

Original comment by `devid...@gmail.com` on 10 Jul 2008 at 9:50

**Comment by GoogleCodeExporter** [comment]
```
In my opinion, a plugin for pidgin would be a better option, but that's just my
personal opinion.
```

Original comment by `ian.haw...@gmail.com` on 14 Jul 2008 at 7:30

**Comment by GoogleCodeExporter** [comment]
```
Are you sure this is in the official client? I thought it was MSN Plus (or 
something)
that did this?
```

Original comment by `sfisher1...@gmail.com` on 17 Jul 2008 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
Yes it was,at first, it was a plugin from MsnPLUS when the client was called 
MSN, but in WLM its built in.
```

Original comment by `SilicioU...@gmail.com` on 17 Jul 2008 at 6:15

**Comment by GoogleCodeExporter** [comment]
```
I think >>  http://trac.adiumx.com/ticket/6578  << got it figure it out, the 
only thing it needs its a developer to 
make it plugin or build it in. Giandrea said that its pretty much done!
```

Original comment by `SilicioU...@gmail.com` on 17 Jul 2008 at 6:17

**Comment by GoogleCodeExporter** [comment]
```
msn-pecan is a protocol not only a plugin for pidgin, so if possible this 
function
should be included in msn-pecan. If it is impossible I hope is possible to drop
bbcode (as [c=4]A[/c]) from nick (example: if nick is [c=4]A[/c] show only A).
```

Original comment by `devid...@gmail.com` on 18 Jul 2008 at 10:01

**Comment by GoogleCodeExporter** [comment]
```
"Yes it was,at first, it was a plugin from MsnPLUS when the client was called 
MSN,
but in WLM its built in."

I'm afraid it isn't - it's still part of MSN Plus (Or Plus Live as it's called 
these
days.) In a stock WLM install, you still see the BBCode and/or colour codes.
```

Original comment by `database404` on 25 Jul 2008 at 11:06

**Comment by GoogleCodeExporter** [comment]
```
"I'm afraid it isn't - it's still part of MSN Plus (Or Plus Live as it's called 
these
days.) In a stock WLM install, you still see the BBCode and/or colour codes."

Anyway, included by default or not, that's a feature that could make the 
contact list
a bit more clear. I agree with "devidfil", if that cannot be included, then 
just drop
the tag. 
(thanks for this plugin)

```

Original comment by `dbk...@gmail.com` on 17 Dec 2008 at 10:05

**Comment by GoogleCodeExporter** [comment]
```
I've started to work on the parser that will drop the tags, it seems to work 
fine:
dfiloni@dfiloni-laptop:~$ ./parser
[c=40]A[/c][c=41]B[/c][i]C[/i]
ABC
dfiloni@dfiloni-laptop:~$ 

The only tags that it drops for now are:
[c=XX]
[/c]
[i]
[/i]

Do they exist? Are there other tags that should be dropped?
```

Original comment by `devid...@gmail.com` on 24 Dec 2008 at 4:54

**Comment by GoogleCodeExporter** [comment]
```
I've just finished to write the code to drop the tags and it works fine, I will 
try
to make it working with pecan. I think will be an option to drop the Plus! tags 
from
nicknames.
```

Original comment by `devid...@gmail.com` on 25 Dec 2008 at 10:13

**Comment by GoogleCodeExporter** [comment]
```
I am using the plus plugin for see color nick like Plus on Windows. I had to 
patch
blist and that use the plugin, but if can be released an msn-pecan that 
integrate
plus/anti-plus I think that many ppl should be happy!
```

Original comment by `mastrop...@gmail.com` on 12 Jan 2009 at 3:15

Attachments:
- [Schermata-Lista contatti.png](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-78/comment-15/Schermata-Lista contatti.png)

**Comment by GoogleCodeExporter** [comment]
```
Yes, in the next version will be an option (I think enabled by default) to hide 
plus
tags from nicknames and personal messages.
```

Original comment by `devid...@gmail.com` on 12 Jan 2009 at 4:38

**Comment by GoogleCodeExporter** [comment]
```
I don't think it needs to be an option. It's closer to what the official client
displays, I doubt anyone will want to show the tags.
```

Original comment by `felipe.contreras` on 12 Jan 2009 at 5:15

**Comment by GoogleCodeExporter** [comment]
```
I think that it should be an option because someone may develop a plugin to show
colors and then the option should be disabled, or if it doesn't work as should 
etc...
I think options are good!
```

Original comment by `devid...@gmail.com` on 12 Jan 2009 at 5:20

**Comment by GoogleCodeExporter** [comment]
```
Well at least if someone use BBCode in their nickname..It's not showing the 
codes..

Example: [b]A[/b] >> only showing A (No bold effect)
```

Original comment by `ongkaq...@gmail.com` on 21 Jan 2009 at 1:40

**Comment by GoogleCodeExporter** [comment]
```
Why make a patch that only remove the tag? We wan't the color too.
```

Original comment by `mastrop...@gmail.com` on 1 Feb 2009 at 1:09

**Comment by GoogleCodeExporter** [comment]
```
mastropino: that requires modification in Pidgin, there's not much msn-pecan 
can do
except fill a bug report in Pidgin.

Once that is done then we can implement it.
```

Original comment by `felipe.contreras` on 1 Feb 2009 at 2:11
- Added labels: **Component-libpurple**

**Comment by GoogleCodeExporter** [comment]
```
Just to let you know, there is some work going on in the related pidgin bug, 
hope it
gets included in mtn, maybe leave a vote :)
http://developer.pidgin.im/ticket/4598
```

Original comment by `markuznightwind@gmail.com` on 21 Jun 2009 at 1:18

**Comment by GoogleCodeExporter** [comment]
```
@markuznightwind thanks for the link. Hopefully they'll merge it, but my guess 
is
that they won't :(

Let's see.
```

Original comment by `felipe.contreras` on 21 Jun 2009 at 7:16

**Comment by GoogleCodeExporter** [comment]
```
Having the option to strip out the color tags is also good if it's easier to
implement (for me at least).
```

Original comment by `vis...@gmail.com` on 8 Aug 2009 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
@viskoo: already done, the option is called "Don't show Messenger Plus! tags"
```

Original comment by `devid...@gmail.com` on 8 Aug 2009 at 7:17

**Comment by GoogleCodeExporter** [comment]
```
Has anyone checked pidgin-plus?
http://polpoinodroidi.com/2009/10/01/pidgin-plus-con-gradienti/
```

Original comment by `felipe.contreras` on 14 Dec 2009 at 12:34

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
The long awaited patch for plus plugin is finally @ mtn :)

Here:
http://bit.ly/9Vxuwf
```

Original comment by `markuznightwind@gmail.com` on 25 Mar 2010 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
That patch only enables a signal. The plugin would still be needed.
```

Original comment by `felipe.contreras` on 27 Mar 2010 at 10:35

**Comment by GoogleCodeExporter** [comment]
```
I believe there's nothing else msn-pecan can do. Somebody needs to implement 
the Plus! plug-in in Pidgin:

http://code.google.com/p/pidgin-plus/

I'll set this as blocked.
```

Original comment by `felipe.contreras` on 14 May 2011 at 8:23
- Changed state: **Blocked**
